app/src/main/java/com/github/sdpteam15/polyevents/view/activity/admin/EventManagementActivity.kt

```diff
@@ -1,13 +1,356 @@
+
+        if (!onCallback) {
+            btnManage.text = getString(R.string.create_event_btn_text)
+            nameET.setText("")
+            descET.setText("")
+            switch.isChecked = false
+            nbpart.visibility = View.INVISIBLE
+            nbpart.setText(emptyPartNb)
+        } else {
+            val event = observableEvent.value!!
+            btnManage.text = getString(R.string.update_event_btn_text)
+            nameET.setText(event.eventName)
+            descET.setText(event.description)
+
+            nbpart.setText(emptyPartNb)
+            nbpart.visibility = View.INVISIBLE
+
+            if (event.isLimitedEvent()) {
+                nbpart.setText(event.getMaxNumberOfSlots()!!.toString())
+                nbpart.visibility = View.VISIBLE
+            }
+            switch.isChecked = event.isLimitedEvent()
+            dateStart.postValue(event.startTime!!)
+            dateEnd.postValue(event.endTime!!)
+            updateTextDate(typeEnd)
+            updateTextDate(typeStart)
+        }
+    }
+
+    private fun manageButtonSetup() {
+        val btnManage = findViewById<Button>(R.id.btnManageEvent)
+        if (isCreation) {
+            btnManage.setOnClickListener {
+                if (verifyCondition()) {
+                    currentDatabase.eventDatabase!!.createEvent(getInformation()).observe(this) {
+                        redirectOrDisplayError(
+                            getString(R.string.event_creation_success),
+                            getString(R.string.event_creation_failed),
+                            it.value
+                        )
+                    }
+                }
+            }
+        } else {
+            btnManage.setOnClickListener {
+                if (verifyCondition()) {
+                    currentDatabase.eventDatabase!!.updateEvents(getInformation()).observe(this) {
+                        redirectOrDisplayError(
+                            getString(R.string.event_update_success),
+                            getString(R.string.failed_to_update_event_info),
+                            it.value
+                        )
+                    }
+                }
+            }
+            currentDatabase.eventDatabase!!.getEventFromId(curId, observableEvent).observe(this) {
+                if (it.value) {
+                    setupViewInActivity(true)
+                } else {
+                    HelperFunctions.showToast(  getString(R.string.failed_get_event_information), this)
+                    startActivity(Intent(this, EventManagementListActivity::class.java))
+                }
+            }
+        }
+    }
+
+    private fun updateTextDate(type: String) {
+        if (typeStart == type) {
+            findViewById<EditText>(R.id.et_start_date).setText(dateStart.value!!.toString())
+        } else {
+            findViewById<EditText>(R.id.et_end_date).setText(dateEnd.value!!.toString())
```

Comment:
Little detail, the formatted date Is a bit weired  2021-05-01T21:57  we can consider changing it a bit for the user by removing the 'T'

Reply:
Yep, I removed it !